### PR TITLE
Update to Amino v0.13.0-rc0

### PR DESCRIPTION
**Pull request by ebuchman**
Updates to Amino v0.13 which updates the way time is encoded.

Also fixes a bug where we weren't sending Version or NextValidatorsHash in the header to the ABCI app

Replaces https://github.com/tendermint/tendermint/pull/2681, for now

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by ebuchman** [comment]
@Liamsi should we cut a rc1 of amino or make the full release so we can update the MarshalBinary -> MarshalBinaryLenPrefixed as well?

**Comment by Liamsi** [comment]
Let's do a 0.13.0 release that includes the MarshalBinary -> MarshalBinaryLenPrefixed, too. 
https://github.com/tendermint/tendermint/pull/2690 contains the changes for this (WIP).

File: state/state.go [hunk 1]
```diff
@@ -140,29 +140,22 @@ func (state State) MakeBlock(
...
-	block.TotalTxs = state.LastBlockTotalTx + block.NumTxs
-
-	block.ValidatorsHash = state.Validators.Hash()
-	block.NextValidatorsHash = state.NextValidators.Hash()
-	block.ConsensusHash = state.ConsensusParams.Hash()
-	block.AppHash = state.AppHash
-	block.LastResultsHash = state.LastResultsHash
-
-	// NOTE: we can't use the state.Validators because we don't
-	// IncrementAccum for rounds there.
```
Comment (ebuchman) [inline state/state.go]: We should add this note about the IncrementAccum to the godoc for the method

File: types/vote_test.go [hunk 2]
```diff
@@ -93,38 +91,36 @@ func TestVoteSignableTestVectors(t *testing.T) {
...
-				0x9, 0x9, 0x0, 0x9, 0x6e, 0x88, 0xf1, 0xff, 0xff, 0xff, // timestamp
+				0x22,                                                                // (field_number << 3) | wire_type
+				0xb, 0x8, 0x80, 0x92, 0xb8, 0xc3, 0x98, 0xfe, 0xff, 0xff, 0xff, 0x1, // timestamp
 				0x32,                                                                               // (field_number << 3) | wire_type
 				0xd, 0x74, 0x65, 0x73, 0x74, 0x5f, 0x63, 0x68, 0x61, 0x69, 0x6e, 0x5f, 0x69, 0x64}, // chainID
 		},
 	}
 	for i, tc := range tests {
-		got, err := cdc.MarshalBinary(tc.canonicalVote)
+		got, err := cdc.MarshalBinaryBare(tc.canonicalVote)
```
Comment (Liamsi) [inline types/vote_test.go]: The `SignBytes` still use `MarshalBinary` (now MarshalBinaryLengthPrefixed). It's a bit weird to change this in the tests only. Also, I actually use the length in the kms, too. Not blocking a release though.

**Pull request merged**